test/local_test_autosplit.py

```diff
@@ -78,107 +67,131 @@ def inspect_split_module(
         print(submod)
 
 
-# Common function to run pipeline with input and check equivalence
-def run_pipe_driver(ec_pipe, args):
-    nstages = len(list(ec_pipe.split_gm.children()))
+# Common function to run pipeline stage with input and check equivalence
+def run_compile_stage(mod: torch.nn.Module, stage: Pipe, args):
+    nstages = len(list(stage.split_gm.children()))
+    ec_input = torch.randn(bs, d_hid, device=args.device)
 
-    pipe_driver: PipelineDriverBase = schedules[args.schedule](
-        ec_pipe,
-        nstages,
-        args.world_size,
-        _debug_mask_minibatches=True,
-        _record_mem_dumps=bool(args.record_mem_dumps),
-        checkpoint=bool(args.checkpoint),
-    )
+    # Warm up and correctness runs
+    if args.rank == 0:
+        stage(ec_input)
+    elif args.rank == args.world_size - 1:
+        out = stage()
+    else:
+        stage()
 
-    # # Warm up and correctness runs
-    ec_input = torch.randn(bs, d_hid, device=args.device)
-    out = pipe_driver(ec_input)
-    ref_out = ec_pipe(ec_input)
+    print(f"Rank {args.rank} completes")
 
     # run with different chunk size to exercise microbatch and scheduling components
-    pipe_driver.chunks = 1
-    pipe_driver(ec_input)
-    pipe_driver.chunks = 100
-    pipe_driver(ec_input)
-
-    if CHECK_NUMERIC_EQUIVALENCE:
-        torch.testing.assert_close(out["out"], ref_out["out"])
-        print(
-            f'equivalence test passed {torch.sum(out["out"])} ref {torch.sum(ref_out["out"])}'
-        )
-
-    # # Profiling runs
+    # stage.chunks = 1
+    # stage(ec_input)
+    # stage.chunks = 100
+    # stage(ec_input)
+
+    # check numeric equivalence in the last rank
+    if args.rank == args.world_size - 1:
+        if CHECK_NUMERIC_EQUIVALENCE:
+            ref_out = mod(ec_input)
+            torch.testing.assert_close(out, ref_out)
+            print(
+                f'equivalence test passed {torch.sum(out["out"])} ref {torch.sum(ref_out["out"])}'
+            )
+
+    # Profiling runs
     with torch.autograd.profiler_legacy.profile(
         enabled=PROFILING_ENABLED
     ) as prof:
-        pipe_driver.chunks = nstages
-        out = pipe_driver(ec_input)
-        ref_out = ec_pipe(ec_input)
-        print(
-            f'profiling run completed {torch.sum(out["out"])} ref {torch.sum(ref_out["out"])}'
-        )
+        stage.chunks = nstages
```

Comment:
I don't think stage supports dynamically changing chunks

Reply:
oh ok, will get rid of that